flojoy/hflib/hub_models/image_caption.py

```diff
@@ -0,0 +1,45 @@
+from enum import Enum, auto
+from typing import NamedTuple
+import transformers as ts
+
+from ..hub_models import HubModel, HubModelFactory
+
+
+class ImageCaptionModels(Enum):
+    NLP_CONNECT_VIT_GPT2 = auto()
+
+
+class ImageCaptionModel(NamedTuple):
+    model: ts.VisionEncoderDecoderModel
+    feature_extractor: ts.ViTImageProcessor
+    tokenizer: ts.AutoTokenizer
+
+
+@HubModelFactory.register(ImageCaptionModels.NLP_CONNECT_VIT_GPT2)
+class NLPConnectVitGPT2(HubModel):
+    def __init__(self):
+        self._model, self._feature_extractor, self._tokenizer = None, None, None
```

Comment:
Do you think it would be a good idea to makes these class attributes instead? I'm thinking about if there were multiple nodes using this model, it would have to download the model once for each node. It would be nicer if it only had to download once. I'm assuming since these are pretrained models they don't have to hold any state, but let me know if I'm wrong!